kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/psi/PsiProvider.kt

```diff
@@ -38,11 +44,34 @@ object PsiProvider {
     }
 
     fun collectPsiMethods(psiFile: PsiFile): MutableCollection<KtNamedFunction> {
-        return PsiTreeUtil.collectElementsOfType(psiFile, KtNamedFunction::class.java)
+        return collectElementsOfType(psiFile, KtNamedFunction::class.java)
+    }
+
+    fun collectPsiImportDirectives(psiFile: PsiFile): MutableCollection<KtImportDirective> {
+        return collectElementsOfType(psiFile, KtImportDirective::class.java)
+    }
+
+    private fun <T : PsiElement> extractElementsOfTypeFromProject(
+        project: Project,
+        psiElementClass: Class<T>
+    ): List<T> {
+        // We should reverse the list with method since we handle all elements separately
+        // and we can invalidate the previous one
```

Comment:
Probably this comment is no longer relevant since it is related to a single case (collecting methods to delete comments from them). 
Maybe we can change method signature to ```extractElementsOfTypeFromProject(
        project: Project,
        psiElementClass: Class<T>,
        reverse: Boolean = false
        )```

Reply:
Move `.reverseed()` and comment to `extractMethodsFromProject()` where it is related to.